lucene/src/main/java/org/hibernate/hql/lucene/internal/builder/LucenePropertyHelper.java

```diff
@@ -73,10 +73,10 @@ private Object convertToPropertyType(String entityType, List<String> propertyPat
 		//Order matters! Some types are subclasses of others
 		//TODO expose something in Hibernate Search so that we can avoid this horrible code
 		if ( bridge instanceof NullEncodingTwoWayFieldBridge ) {
-			return convertToPropertyType( entityType, propertyPath, value, ( (NullEncodingTwoWayFieldBridge) bridge ).unwrap() );
+			return convertToPropertyType( entityType, propertyPath, value, ( (NullEncodingTwoWayFieldBridge) bridge ).unwrap( NullEncodingTwoWayFieldBridge.class ) );
```

Comment:
This will most likely result in a null unwrapped bridge, since BridgeAdaptor.unwrap doesn't take the adaptor itself into account. 

Anyway... here too, can't we simply use the metadata? Add an abstract `getNumericEncoding` method instead of introspecting field bridges? It would seem simpler and easier to maintain...